addons/portal/wizard/portal_invite_share.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models, _
+
+
+class PortalInviteShare(models.TransientModel):
+    _name = 'portal.invite.share'
+
+    @api.model
+    def default_get(self, fields):
+        result = super(PortalInviteShare, self).default_get(fields)
+        result['res_model'] = self._context.get('active_model')
+        result['res_id'] = self._context.get('active_id')
+        result['share_link'] = self.env[result['res_model']].browse(result['res_id'])._get_share_url()
+        return result
+
+    res_model = fields.Char('Related Document Model', required=True)
+    res_id = fields.Integer('Related Document ID', required=True)
+    partner_ids = fields.Many2many('res.partner', string="Recipients", required=True)
+    note = fields.Text(help="Add extra content to display in the email")
+    share_link = fields.Char(string="Link", compute='_compute_share_link')
```

Comment:
Actually as it is recomputed in the send process, is this field necessary ?

Reply:
indeed !